l1-contracts/test/foundry/integration/_SharedHyperchainDeployer.t.sol

```diff
@@ -0,0 +1,132 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.24;
+
+import {L1ContractDeployer} from "./_SharedL1ContractDeployer.t.sol";
+import {RegisterHyperchainsScript} from "./deploy-scripts/script/RegisterHyperchains.s.sol";
+import {ETH_TOKEN_ADDRESS} from "contracts/common/Config.sol";
+
+contract HyperchainDeployer is L1ContractDeployer {
+    RegisterHyperchainsScript deployScript;
+    HyperchainDeployInfo[] hyperchainsToDeploy;
+
+    struct HyperchainDeployInfo {
+        string name;
+        RegisterHyperchainsScript.HyperchainDescription description;
+    }
+
+    uint256 currentHyperChainId = 10;
+    uint256 eraHyperchainId = 9;
+    uint256[] public hyperchainIds;
+
+    function deployHyperchains() internal {
+        deployScript = new RegisterHyperchainsScript();
+
+        hyperchainsToDeploy.push(getDefaultHyperchainDeployInfo("era", eraHyperchainId, ETH_TOKEN_ADDRESS));
+        hyperchainIds.push(eraHyperchainId);
+
+        saveHyperchainConfig();
+
+        vm.setEnv(
+            "HYPERCHAINS_CONFIG",
+            "/test/foundry/integration/deploy-scripts/script-out/output-deploy-hyperchains.toml"
+        );
+
+        deployScript.run();
+    }
+
+    function addNewHyperchainToDeploy(string memory _name, address _baseToken) internal {
+        hyperchainsToDeploy.push(getDefaultHyperchainDeployInfo(_name, currentHyperChainId, _baseToken));
+        hyperchainIds.push(currentHyperChainId);
+        currentHyperChainId++;
+    }
+
+    function getDefaultDescription(
+        uint256 __chainId,
+        address __baseToken
+    ) internal returns (RegisterHyperchainsScript.HyperchainDescription memory description) {
+        description = RegisterHyperchainsScript.HyperchainDescription({
+            hyperchainChainId: __chainId,
+            baseToken: __baseToken,
+            bridgehubCreateNewChainSalt: 0,
```

Comment:
Worth to test it too?

Reply:
What exactly? Whole contract or specific thing? Because right now it's only used in test deploy script, not actual deploy scripts.

Reply:
`bridgehubCreateNewChainSalt`